dask/dataframe/groupby.py

```diff
@@ -537,12 +571,17 @@ def _meta_nonempty(self):
         """
         Return a pd.DataFrameGroupBy / pd.SeriesGroupBy which contains sample data.
         """
+        # TODO: unify with index logic
         sample = self.obj._meta_nonempty
         if isinstance(self.index, Series):
             if self._is_grouped_by_sliced_column(self.obj, self.index):
                 grouped = sample.groupby(sample[self.index.name])
             else:
                 grouped = sample.groupby(self.index._meta_nonempty)
+        elif (isinstance(self.index, list) and
+              any(isinstance(item, Series) for item in self.index)):
+            # TODO: implement this
+            raise NotImplementedError("raise an error to prevent a segfault")
```

Comment:
I haven't checked yet (working down through code) but it would be good to have an xfailed test for this if you don't have one already.